fip-6.md

```diff
@@ -0,0 +1,141 @@
+##### Request
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_domain|Yes|FIO Domain, see FIO Domain validation rules.|FIO domain that is being requested to burn.|
+|actor|Yes|FIO account name|FIO account for the signer, the account owning this FIO Domain.|
+|max_fee|Yes|max fee SUFs|Maximum amount of SUFs the user is willing to pay for fee. Should be preceded by /get_fee for correct value.|
+|tpid|Yes|FIO Address of TPID, See FIO Address validation rules|FIO Address of the wallet which generates this transaction. This FIO Address will be paid 10% of the fee.See FIO Protocol#TPIDs for details. Set to empty if not known.|
+
+###### Example
+```
+{
+    "fio_domain": "alice",
+    "actor": "aftyershcu22",
+    "max_fee": 800000000,
+    "tpid": "rewards@wallet"
+}
+```
+##### Processing
+* Require auth of the actor.
+* Request is validated per Exception handling.
+* Sets fio_domain expiration date inside the domains table to now().
+* Return status json containing status (ok), and fee charged.
+##### Response
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid FIO Domain format|FIO Domain format is not valid|400|"fio_domain"|Value sent in, e.g. "alice"|"Invalid FIO domain"|
+|FIO Domain not registered|FIO Domain is not registered|400|"fio_domain"|Value sent in, e.g. "alice"|"FIO Domain not registered"|
+|Not owner of FIO Domain|The signer does not own the domain|403||||
+|Invalid fee value|max_fee format is not valid|400|"max_fee"|Value sent in, e.g. "-100"|"Invalid fee value"|
+|Insufficient funds to cover fee|Account does not have enough funds to cover fee|400|"max_fee"|Value sent in, e.g. "800000000"|"Insufficient funds to cover fee"|
+|Invalid TPID|tpid format is not valid|400|"tpid"|Value sent in, e.g. "notvalidfioaddress"|"TPID must be empty or valid FIO address"|
+|Fee exceeds maximum|Actual fee is greater than supplied max_fee|400|max_fee"|Value sent in, e.g. "800000000"|"Fee exceeds supplied maximum"|
+##### Response
+|Parameter|Format|Definition|
+|---|---|---|
+|status|String|Ok|
+|fee_collected|String|fee amount collected SUFs|
+###### Example
+```
+{
+    "status": "Ok",
+    "fee_collected": "800000000"        
+}
+```
+## Fees
+A new fee will be created for `deactivate_fio_domain` . This fee will be of type 0 and should cost 800000000 SUF.
+
+## Rationale
+Some users might find it useful to have the ability to deactivate unused addresses or domains. This feature proposal will allow Address owners the ability to burn their addresses at any time. Domain owners that are looking to deactivate will have their expiration set to the current time. Setting this expiration allows for the continued use of the domain during a grace period until the new owner registers. If the domain does not receive a new owner in time, all FIO Addresses linked to that domain will be burned after their expirations but will not be able to be utilized.
```

Comment:
What is the grace period? Does the grace period start immediately, regardless of how far into the future the domain was paid up?

As to burning all FIO addresses, can you outline a bit more on how the grace period impacts them, how they will know the domain they are on needs to be paid for, etc? What happens when someone buys up the domain and it's transferred to a new owner (nothing?)? If an individual address user can pay the fee for the domain in order to keep their address functional, that would be good to highlight here. Also, people may be unwilling to pay for addresses if they can't know for sure if the domain will be around in the future.

Reply:
Currently, `DOMAINWAITFORBURNDAYS` and `ADDRESSWAITFORBURNDAYS` are set inside fio.common at 90 days.

Reply:
@lukestokes interesting idea on the ability to pay the Domain fee before it is burned (but after it is in the expired grace period) but that could be strange as someone could take over the hosting of the domain your address is on without you knowing it. As an FYI, the full details on domain expiration are at: https://stealth.atlassian.net/wiki/spaces/DEV/pages/90898460/FIO+Protocol#FIOProtocol-ExpiredDomain
I think the idea is that certain address and domain functions will stop working when a domain is expired and that will serve as notification that something is wrong (they will return an error "400:FIO Domain expired").

Reply:
Could help to change the effected calls to return "400: FIO Domain expired or deactivated"